### DISCUSSION TITLE
Googlebot and mixed contributor PRs

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#18776] [Request Help]
I need to determine if `goooglebot` still enforces this rule:

<img width="920" alt="image" src="https://user-images.githubusercontent.com/6311784/200109071-2aff3d23-2686-4acc-8fd0-680ecc11066c.png">

I don't find *recent* examples of this message in github.com.

Currently Renovate uses `@renovate-bot` as both commit author and PR creator so that the commit and PR come from the same account, but I would like to move to a model where the Forking Renovate app submits the PRs again.

@bcoe would be able to confirm this or help me find it out? Thanks

**Comment by rarkins** [comment]
I see that it's now `google-cla` doing the enforcement and not `googlebot`. 

More recent example: https://github.com/google/android-fhir/pull/326#issuecomment-830116152

It seems that the problem still occurs *but* it seems to give the option of `cla:yes` label. Example repo: https://github.com/google/benchmark/issues?q=is%3Aopen+label%3A%22cla%3A+yes%22+sort%3Aupdated-desc

I'm not sure if this is valid for all Google repos or if it's OK for Renovate to cla approve its PRs using a label? If Renovate set the CLA label to yes, would the bot reset it to no?

**Reply by rarkins** [reply]
On that same repo I now see the CLA labels not in use. A recent PR shows this message:

<img width="858" alt="image" src="https://user-images.githubusercontent.com/6311784/200109600-b6cebbbe-ac31-46ae-a953-060d03b9e622.png">

**Comment by rarkins** [comment]
Another recent example:

<img width="1091" alt="image" src="https://user-images.githubusercontent.com/6311784/200109720-ab800786-bb6d-452b-b3fe-e00118dbb1b4.png">

From https://github.com/GoogleCloudPlatform/microservices-demo/runs/5396785624

**Comment by rarkins** [comment]
I have experimented with a PR which has a [failed check](https://github.com/google/js-green-licenses/pull/191/checks) for the bot:

<img width="432" alt="image" src="https://user-images.githubusercontent.com/6311784/200113960-6a9fc751-35f4-47df-b587-e04e054d655f.png">

Reading https://opensource.google/documentation/reference/cla#external_bots I see:

<img width="887" alt="image" src="https://user-images.githubusercontent.com/6311784/200114033-05a93b1a-20a2-4d1e-832c-e66fea0bdd3e.png">

I'm guessing this refers to:

<img width="887" alt="image" src="https://user-images.githubusercontent.com/6311784/200114067-867f8f63-5d3f-46d5-861a-e6c34f8f973f.png">

I would like to submit for the following usernames to be approved:
- `forking-renovate[bot]`
- `renovate[bot]`
- `mend[bot]`

The latter two accounts will be used for *committing*, while the first will be used for PR creation.

**Comment by bcoe** [comment]
Hey @rarkins 👋 

Looping in @rspier who's a CLA Bot expert.

**Comment by rspier** [comment]
Hi @bcoe,

   CLAs aren't in scope for me anymore.  But I'll get you to the right place:

   Please file an internal request via go/cla#links so the proper process can happen.

-R

**Reply by rarkins** [reply]
@rspier unfortunately I'm not internal! Any chance you know someone else who can help me through the process or you could kick it off so that I can find the right person?

**Reply by rspier** [reply]
Hi @rarkins.  There's not really anything for you to do.  There's an analysis to verify that the tool only does safe operations (the bits you highlighted above) and if there are no issues/concerns it gets added to a configuration. I was hoping @bcoe would file the internal request.  If he doesn't want to, I can do it.
-R

**Reply by rarkins** [reply]
Thanks. What I want to do is get the bot names "pre-approved" in advance, rather than temporarily break the checks when we switch. The bots themselves have same functionality as today, it's just about an identity change.

**Reply by rspier** [reply]
That wasn't clear from the above.  If it's just a name change, that's much simpler.  (I'll look in the morning if I have a chance.)

**Reply by rarkins** [reply]
Sorry for that, and thanks for your time

**Reply by rspier** [reply]
`mend[bot]` doesn't seem to exist yet.  We need to record its `id`.  https://api.github.com/users/mend[bot]

I've sent off a change to add them both to the allowlist, but notated not to submit it until we figure out `mend[bot]`'s id.  It's not a big deal to do it as two different changes, but cleaner to do it as one.  Once it's submitted, it'll take effect the next (Monday to Thursday) late morning California time.

**Reply by rarkins** [reply]
Hi, in this case please go ahead without `mend[bot]`. This was me planning ~6 months ahead where we might rename one of the bots but even in that case it would have an existing ID. In such case would you be able to submit as-is? Thanks again for your time

**Reply by rspier** [reply]
It's submitted.    It should be live by noon (California time) on Monday.

**Reply by rspier** [reply]
Did you confirm if it's now working for you?

**Reply by rarkins** [reply]
I was a little hesitant to put it live already, sorry! I will do so this week so that PRs show up as coming from the bot account instead of user account as today

**Reply by rspier** [reply]
No rush at all. I just didn't want an uncaught typo or something we missed to sneak up later.

**Reply by rarkins** [reply]
Thanks again. FYI I just discovered a "hitch" which prevents me rolling out this PR creator change right away. Right now we are able to enable the "maintainers can modify" GitHub feature on PRs, which is quite convenient. Unfortunately GitHub only allows this feature to be enabled if:
- The forked repo is owned by a user, not an org, and
- The PR is created by the owner of the repo

Therefore if we were to switch to having the bot create PRs, this option is no longer available. We'll need to consider alternatives before making the switch.
